### PR TITLE
Port `translate-genbank-location` to `augur curate parse-genbank-location` [#1485]

### PR DESCRIPTION
**Pull request by genehack**
## Description of proposed changes

This ports `translate-genbank-location` from nextstrain/ingest to the `augur curate` sub-command `parse-genbank-location`.

It adds a command-line flag for providing the field name the location data is stored in; this flag defaults to `geo_loc_name` to support NCBI's recent change. I provided this to enable the command to also parse older data with the `country` field name. 

Also adds type-hints and tests. 

## Related issue(s)

* #1485 
* https://github.com/nextstrain/ingest/issues/43

## Checklist

- [ ] Checks pass
- [x] If making user-facing changes, add a message in [CHANGES.md](https://github.com/nextstrain/augur/blob/HEAD/CHANGES.md) summarizing the changes in this PR

### CODE REVIEW

**Comment by genehack** [comment]
The one failing test is [expected](https://github.com/nextstrain/conda-base/issues/75).

File: CHANGES.md [hunk 1]
```diff
@@ -1506,9 +1506,9 @@
   [Also part of PR 431](https://github.com/nextstrain/augur/pull/431)
 * traits: Allow input of `--weights` which references a `.tsv` file in the following format:
   ```
-  division	Hubei	10.0
-  division	Jiangxi	1.0
-  division	Chongqing	1.0
+  division  Hubei   10.0
```
Comment (jameshadfield) [inline CHANGES.md:1509]: I think these tab-to-spaces conversions are automatic from your editor, right? They shouldn't be part of this commit (also some other unintended whitespace changes on lines 1345-1347)

Comment (genehack) [reply CHANGES.md:1509]: Ah geez, didn't see those, sorry. Thanks for catching.

File: augur/curate/parse_genbank_location.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    return parser
+
+
+def run(
+    args: argparse.Namespace,
+    records: List[dict],
+) -> Generator[dict, None, None]:
+    for record in records:
+        database = record.get("database", "")
+        if database in {"GenBank", "RefSeq"}:
```
Comment (joverlee521) [inline augur/curate/parse_genbank_location.py:73]: _non-blocking, for discussion only_

A part of me just wants to remove the `database` check and let people use the command at their own discretion...

**Review by joverlee521 — APPROVED** [review]

File: augur/curate/parse_genbank_location.py (hunk 2 above)
Comment (genehack) [reply augur/curate/parse_genbank_location.py:73]: I mean, what's the downside to that?

Comment (joverlee521) [reply augur/curate/parse_genbank_location.py:73]: This was originally written for mpox, where at one point we had a mix of data sources going through the same curation pipeline. This database check allowed the script to skip records that were not GenBank records. 

So I guess the downside is people would not be able to use the same curation pipeline for multiple data sources...

Comment (genehack) [reply augur/curate/parse_genbank_location.py:73]: I mean, it's going to no-op if the record doesn't have the `geo_loc_name` field (or whatever `--location-field` is set to); did the non-GenBank records have that field? Seems to me like duck-typing on the field you're operating on is a better check than keying off a second field, but I am but an egg here.

Comment (joverlee521) [reply augur/curate/parse_genbank_location.py:73]: Tracking in https://github.com/nextstrain/augur/issues/1511

Comment (genehack) [reply augur/curate/parse_genbank_location.py:73]: yep, just grabbed that. probably won't get to it until tomorrow tho.
